### ISSUE TITLE
Improve column toggling UX

### ISSUE THREAD

**Issue by icflorescu** [icflorescu/mantine-datatable#518]
I think there's some room for improvement in the column toggling UX.
In my opinion, the toggling menu should be on the header instead of the toggleable columns.

That way, the user would not experience "jumps" when toggling columns using the menu, and we'd have fewer components and DOM elements created, so the entire code would probably be more efficient.

What do you think, @gfazioli?

![image](https://github.com/icflorescu/mantine-datatable/assets/581999/d98646de-193c-4ba1-a528-1b42cfb6d638)

**Comment by icflorescu** [comment]
Hey, @gfazioli!
Related to the above, please have a look at this branch when you have a moment:
https://github.com/icflorescu/mantine-datatable/tree/fix/518-improve-column-toggling-ux  

Basically, in [this commit](https://github.com/icflorescu/mantine-datatable/commit/5f0e78b96a359adfdd91cf76fb43eb5af3dfdacf) I've moved the column toggling popover on the table header.

You can test it locally in this page: http://localhost:3000/examples/column-dragging-and-toggling/#column-toggling

It looks to me like it's working perfectly, but since you're the author of the feature, I thought I should ask for your opinion, in case there are hidden issues I'm not aware of.

Thanks again for everything!

**Comment by gfazioli** [comment]
> It looks to me like it's working perfectly, but since you're the author of the feature, I thought I should ask for your opinion, in case there are hidden issues I'm not aware of.

Hey, I just checked and it works like a charm for me too. I think this way is definitely better. Well done 👍 

btw, I would like to discuss [this enhancement](https://github.com/icflorescu/mantine-datatable/issues/476) as it could be useful for column management. wdyt?

**Comment by gfazioli** [comment]
> I just published it in `v7`, but I'm not sure if/when I'll find the time to backport it to `v6`.

No problem, if I can, I'll try to add it to the v6 app as soon as possible. Good job!

> I'd be reluctant, however, to do any kind of search logic inside the DataTable component itself because in real-life projects this is usually done on the server-side.

I agree. I think we need to carefully consider features like this. A good library should provide only the useful tools to easily implement any custom feature.
For example, in the case of the header, I believe it should generally be tailored specifically for each project, as there are truly numerous combinations of different needs.

If we try to imagine a possible implementation of a Header, since we now have column management (resetting the order, visibility, and size), we would have something like this.

```
+------------------------------+---------+
| {title}                      | {tools} |
+------------------------------+---------+
```
- We should plan to choose a title, show it, and support a `string` or React component
- The same for the tools, how to visualize them along with alignment, etc.

Any eventual filter would not have much meaning in my opinion, because it depends on many factors; what is being displayed, if pagination is active, etc.

Eventually, what could be done is a simple "slot" attribute to insert a custom component, for example.

```tsx
<DataTable header={<MyHeader />} />
```

which in the end would be essentially the same as

```tsx
<MyHeader />
<DataTable />
```

In short, at the moment I can't think of any ideas on how to generalize a header without overly complicating the code.
It would perhaps be more useful to show how to create a custom header, with some examples and/or external components.

Below, for example, I have implemented a custom header on the application side.
<img width="993" alt="image" src="https://github.com/icflorescu/mantine-datatable/assets/432181/1475efd8-a52a-46d4-8819-97ec73889012">

<img width="1066" alt="image" src="https://github.com/icflorescu/mantine-datatable/assets/432181/0e0b3ece-ebd9-4855-829b-4c94ba69f166">

![image](https://github.com/icflorescu/mantine-datatable/assets/432181/157f104a-2381-4f63-b6b7-dfbb4967e887)
